roles/deploy_user/tasks/main.yml

```diff
@@ -12,11 +12,37 @@
     home: "/home/{{ deploy_user }}"
     shell: "{{ deploy_user_shell }}"
 
-- name: deploy_user | add SSH keys
-  ansible.posix.authorized_key:
-    user: "{{ deploy_user }}"
-    key: "{{ item.key }}"
-  with_items: "{{ deploy_ssh_users }}"
+- name: deploy_user | get key content from github
+  ansible.builtin.command: 'curl {{ item }}'
+  loop: "{{ deploy_user_github_keys }}"
+  register: deploy_user_keys_from_github
+  changed_when: false
+
+- name: deploy_user | create the .ssh directory
+  ansible.builtin.file:
+    path: "/home/{{ deploy_user }}/.ssh/"
+    state: directory
+    owner: "{{ deploy_user }}"
+    group: "{{ deploy_user }}"
+    mode: 0700
+  when: not running_on_server
+
+- name: deploy_user | build authorized keys file
+  ansible.builtin.template:
+    src: authorized_keys.j2
+    dest: /home/{{ deploy_user }}/.ssh/authorized_keys
+    group: "{{ deploy_user }}"
+    owner: "{{ deploy_user }}"
+    mode: '0600'
+    backup: true
+
+# - name: deploy_user | add SSH keys
```

Comment:
Is this needed?

Reply:
I think the commented task can go - I meant to remove it and just never got around to it.

@kayiwa remember that this is for the `deploy` user, so it's less urgent than for our usual SSH user. Also, the new task sets `backup: true` so when the `authorized_keys` file changes, we will always have the previous file available on the VMs.

Reply:
addressed in 5691a35087107bf8fe04c3b967b4e5bfaf537aec